gems/aws-sdk-core/spec/aws/plugins/jsonvalue_converter_spec.rb

```diff
@@ -109,7 +109,8 @@ module Plugins
         }
         ruby_types.each do |_k, v|
           resp = client.foo(header_field: v)
-          expect(resp.context.params[:header_field]).to eq(v.to_json)
+          expected_value = v.is_a?(String) ? v : v.to_json
```

Comment:
I'm not 100% certain on this behavior anymore.  This definitely passes the protocol tests as we have them written, but technically does not encode json strings correctly.

I think we may instead need to modify our protocol test runner to handle this case differently - specifically - if we see a string param for an input that is marked with the jsonvalue trait, then we do a parse on it.  I think worth asking Kevin about to confirm what the behvior here should be.

Reply:
See my comment [here](https://github.com/aws/aws-sdk-ruby/pull/2988/files#r1550748493) on resolution